plugins/modules/system/pamd.py

```diff
@@ -733,14 +733,19 @@ def __str__(self):
         lines = []
         current_line = self._head
 
+        mark = "# Updated by Ansible - %s" % datetime.now().isoformat()
         while current_line is not None:
             lines.append(str(current_line))
             current_line = current_line.next
 
-        if lines[1].startswith("# Updated by Ansible"):
-            lines.pop(1)
-
-        lines.insert(1, "# Updated by Ansible - " + datetime.now().isoformat())
+        if len(lines) <= 1:
+            lines.insert(0, "")
```

Comment:
Shouldn't this only be inserted if `len(lines) == 0`?

Reply:
(Since you mentioned above that `len(lines) == 0` does not happen, how about simply removing this line, and changing the next one to `lines.append(mark)`?)

Reply:
> > Shouldn't this only be inserted if `len(lines) == 0`?
> 
> Not really. The whole point of the function here is to insert this timestamped mark _before_ the rules - and we want it to be in the second line always, so that we need less checks below, so adding a blank and the timestamp works for both cases 0 and 1.

But if `len(lines) == 1`, then your code will put the mark in the third line and not in the second line!

Reply:
??
```
0: session required pam_lastlog.so silent showfailed
```
=> insert 0 ""
```
0: 
1: session required pam_lastlog.so silent showfailed
```
=> insert 1 "# Updated by ..."
```
0:
1: # Updated by ...
2: session required pam_lastlog.so silent showfailed
```
?

Reply:
Meh, I've been totally blind :) You're right!